### PR TITLE
Bug 1505955 - Replace PhFramework with getData

### PR DESCRIPTION
**Pull request by ionutgoldan**

### CODE REVIEW

**Comment by sarah-clements** [comment]
I checked out your branch and saw this message in the console after clicking the Alerts tab: TypeError: "_helpers_http__WEBPACK_IMPORTED_MODULE_5__.getData.getApiUrl is not a function", so it might be that you can only use angulars $http service for fetching data. I'd do a google search to verify and if that's the case, just skip these for now. We can remove them as we go.

**Comment by ionutgoldan** [comment]
> I checked out your branch and saw this message in the console after clicking the Alerts tab: TypeError: "helpers_http__WEBPACK_IMPORTED_MODULE_5_.getData.getApiUrl is not a function", so I'd google it and also check the AngularJS documentation.

I made a silly runtime programming error: I called getData.getApiUrl(...) in the PhAlerts factory from ui/js/models/perf/alerts.js, instead of getData(getApiUrl(...)). This fixed most of the Selenium tests.
And I also had to add a $scope.$digest() call to finish the remaining failing Selenium test.

**Comment by ionutgoldan** [comment]
> EDIT: @ionutgoldan I've realized that type error is because of how you're chaining getData (which uses async/await) with then.(). I'd remove .then() since a Promise is not being returned and use async/await (to wait for the resolved getData response to return).

AFAIK getData() implicitly returns a promise. So calling then() on it shouldn't cause problems. Still, if it's better to use the new await syntax, I could do that.

> Also, you can name the url paths (ex /performance/framework) and move them to the /perfherder/constants.js file.

Done.

File: ui/perfherder/constants.js [hunk 1]
```diff
@@ -1,2 +1,8 @@
 export const tValueCareMin = 3; // Anything below this is "low" in confidence
 export const tValueConfidence = 5; // Anything above this is "high" in confidence
+
+// Backend server endpoints
+export const endpoints = Object.freeze({
+    issueTrackers: '/performance/issue-tracker/',
```
Comment (sarah-clements) [inline ui/perfherder/constants.js]: These can just be declared as `const`s as per my comment in your other pr.

**Comment by ionutgoldan** [comment]
> can you also tag Ed or Camd for final review for all your pull requests?

Yes, will do this from now on.

File: ui/perfherder/constants.js [hunk 2]
```diff
@@ -1,2 +1,8 @@
 export const tValueCareMin = 3; // Anything below this is "low" in confidence
 export const tValueConfidence = 5; // Anything above this is "high" in confidence
+
+// Backend server endpoints
+export const endpoints = {
+    issueTrackers: '/performance/issue-tracker/',
```
Comment (edmorley) [inline ui/perfherder/constants.js]: Could you switch this to 2 space indent please (about to be enforced by the linter in #4276 for everything that's not in the legacy `ui/js/` directory)

**Comment by edmorley** [comment]
Please rebase on master before merging, to ensure eslint passes after the new rules have been enabled :-)

**Comment by ionutgoldan** [comment]
> Please rebase on master before merging, to ensure eslint passes after the new rules have been enabled :-)

Done.

**Pull request merged**